packages/e2e-tests/specs/block-transforms.test.js

```diff
@@ -0,0 +1,175 @@
+		],
+		( blockName ) => parsedBlockObject.blockName === blockName
+	);
+};
+
+const setPostContentAndSelectBlock = async ( content ) => {
+	await setPostContent( content );
+	const blocks = await getAllBlocks();
+	const clientId = blocks[ 0 ].clientId;
+	await page.click( '.editor-post-title .editor-post-title__block' );
+	await selectBlockByClientId( clientId );
+};
+
+const getTransformStructureFromFile = async ( fileBase ) => {
+	if ( isAnExpectedUnhandledBlock( fileBase ) ) {
+		return null;
+	}
+	const { file: content } = getBlockFixtureHTML( fileBase );
+	await setPostContentAndSelectBlock( content );
+	const block = ( await getAllBlocks() )[ 0 ];
+	const availableTransforms = await getAvailableBlockTransforms();
+	const originalBlock = await getBlockTitle( block.name );
+
+	return {
+		availableTransforms,
+		originalBlock,
+		content,
+	};
+};
+
+const getTransformResult = async ( blockContent, transformName ) => {
+	await setPostContentAndSelectBlock( blockContent );
+	expect( await hasBlockSwitcher() ).toBe( true );
+	await transformBlockTo( transformName );
+	return getEditedPostContent();
+};
+
+describe( 'test transforms', () => {
+	// Todo: Remove the filter as soon as all fixtures are corrected,
+	// and its direct usage on the editor does not trigger errors.
+	// Currently some fixtures trigger errors (mainly media related)
+	// because when loaded in the editor,
+	// some requests are triggered that have a 404 response.
+	const fileBasenames = filter(
+		getAvailableBlockFixturesBasenames(),
+		( basename ) => (
+			! some(
+				[
+					'core__image',
+					'core__gallery',
+					'core__video',
+					'core__file',
+					'core__media-text',
+					'core__audio',
+					'core__cover',
+				],
+				( exclude ) => basename.startsWith( exclude )
+			)
+		)
+	);
+
+	const transformStructure = {};
+	beforeAll( async () => {
+		await createNewPost();
+
+		for ( const fileBase of fileBasenames ) {
+			const structure = await getTransformStructureFromFile(
+				fileBase
+			);
+			if ( ! structure ) {
+				continue;
+			}
+			transformStructure[ fileBase ] = structure;
+			await setPostContent( '' );
+			await page.click( '.editor-post-title .editor-post-title__block' );
+		}
+	} );
+
+	it( 'should contain the expected transforms', async () => {
+		const transforms = mapValues(
+			pickBy(
+				transformStructure,
+				( { availableTransforms } ) => availableTransforms,
+			),
+			( { availableTransforms, originalBlock } ) => {
+				return { originalBlock, availableTransforms };
+			}
+		);
+		expect(
+			transforms
+		).toEqual( EXPECTED_TRANSFORMS );
+	} );
+
+	describe( 'individual transforms work as expected', () => {
+		beforeAll( async () => {
+			await createNewPost();
+		} );
+
+		beforeEach( async () => {
+			await setPostContent( '' );
+			await page.click( '.editor-post-title .editor-post-title__block' );
+		} );
+
+		for ( const [ fixture, { originalBlock, availableTransforms } ] of Object.entries( EXPECTED_TRANSFORMS ) ) {
```

Comment:
There is `describe.each` API method, which you might also want to use here: https://jestjs.io/docs/en/api#describeeachtable-name-fn-timeout

It wouldn't have to be expressed using all those loops.